### DISCUSSION TITLE
Mass operations on items

### DISCUSSION THREAD

**Discussion by wientanz** [mealie-recipes/mealie#3682] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Currently there is no evident way of performing mass operations, i.e. I can‘t select multiple recipes to move to cookbooks in one swoop, which is especially a hassle when you do a mass import with hundreds of recipes. Or mass deleting recipes.

### Please provide a concise description of the feature that would resolve your issue.

Implement a checkbox for selection of multiple items and then allow for mass operations, i.e. moving or deleting items

### Please consider and list out some caveats or tradeoffs made in your design decision

None indentified

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
> I can‘t select multiple recipes to move to cookbooks in one swoop

Have you tried the data management page? It can do this.
https://demo.mealie.io/group/data/foods

**Reply by wientanz** [reply]
I did not see this… thanks and sorry for me being blind

**Reply by tw258** [reply]
Thanks @michael-genson ! I did not find this page either. Is it only accessible via directlink (/group/data/foods) or is there a way to get there from the admin section?

**Reply by Kuchenpirat** [reply]
This is not an admin setting as everybody with the "User can organize group data" enabled is able to manipulate this data. 

It can be found by clicking on your profile pic in the top left and then by clicking on the "Manage Data" Section

**Reply by tw258** [reply]
Thank you!

**Reply by wientanz** [reply]
Ok, found it. But my point still stands... I can't send multiple recipes to a cookbook. I can add tags, granted It would be great to be able to send a batch of recipes to a specific cookbook, maybe even during mass import.

**Reply by michael-genson** [reply]
Recipes are added to cookbooks via tags and categories, so you need to apply the correct tags/categories that you've defined on your cookbooks

**Reply by disconn3ct** [reply]
Is there a way to filter that list? For example, if we create a new tag 'taco', I'd like to avoid manually paging through about 1200 recipes to add it. (And if I can avoid scripting the API, I can enlist helpers.)

**Comment by wientanz** [comment]
Ok, thank you for clarifying that. Will try it out. One more question: When
importing single recipes I can have them tagged automatically. Using mass
import I did not see that. Any way to retroactively do that?

Am Mo., 3. Juni 2024 um 15:54 Uhr schrieb Michael Genson <
***@***.***>:

**Reply by michael-genson** [reply]
I don't believe so, you can manually add categories/tags to a bulk import, but there's no way to add them automatically like single recipes (a feature request would be welcome for that!)

**Comment by garymoon** [comment]
I found this via Google just now. The functionality in here is :exploding_head: :heart_eyes: My 2c would be that it's definitely worth surfacing more prominently :100: 

Thank you for Mealie! :blue_heart:
